packages/components/src/button-group/index.tsx

```diff
@@ -20,13 +20,15 @@ function UnforwardedButtonGroup(
 	props: WordPressComponentProps< ButtonGroupProps, 'div', false >,
 	ref: ForwardedRef< HTMLDivElement >
 ) {
-	const { className, ...restProps } = props;
+	const { className, __shouldNotWarnDeprecated, ...restProps } = props;
 	const classes = clsx( 'components-button-group', className );
 
-	deprecated( 'wp.components.ButtonGroup', {
-		since: '6.8',
-		alternative: 'wp.components.ToggleGroupControl',
-	} );
+	if ( ! __shouldNotWarnDeprecated ) {
```

Comment:
Adds a mechanism to suppress a confusing "double warning" coming from the ButtonGroup used internally in RadioGroup.

Reply:
React uses global variables for this purpose:
```js
let didWarnAboutSomething = false;
function something() {
  if ( ! didWarnAboutSomething ) {
    didWarnAboutSomething = true;
    console.error( /* ... */ );
  }
}
```
They also use `Set`s as `didWarn*` values, when they want to track a warning for each component name or something else separately.

We could use this, too, instead of the semi-official prop.

Reply:
If we decide to go the `didWarn` way, I think it makes sense to implement it directly in the `deprecated()` helper function.

Reply:
I was thinking about smarter ways to implement this mechanism when I started logging the size deprecation warnings, but came to the conclusion that an explicit prop was the only way. I am imagining consumer code like this, where the ButtonGroup warning inside the RadioGroup should be suppressed, but the standalone ButtonGroup instance should still warn:

```jsx
const MyUI = () => (
  <RadioGroup />
  <ButtonGroup />
);
```

I'm all ears if you have any clever ideas!

Reply:
Could component's internal context system help?

Reply:
Not in a way that's "more automatic" than the current manual prop-based system, I don't think. Also context could be problematic when render props or slots are involved (e.g. we don't want to suppress warnings on a component that the consumer passed in through a `prefix` prop).